pkg/test/resourcefixture/testdata/basic/sql/v1beta1/sqlinstance/sqlinstance-ssl/update.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: sql.cnrm.cloud.google.com/v1beta1
+kind: SQLInstance
+metadata:
+  name: sqlinstance-ssl-${uniqueId}
+  annotations:
+    cnrm.cloud.google.com/state-into-spec: absent
+spec:
+  databaseVersion: POSTGRES_16
+  region: us-central1
+  settings:
+    ipConfiguration:
+      sslMode: ENCRYPTED_ONLY
+      # When `sslMode=ENCRYPTED_ONLY`, the legacy `requireSsl` flag must be false or cleared.
+      requireSsl: false
```

Comment:
> What happens if
> 
> * `ssLmode: ENCRYPTED_ONLY` and `requireSsl: true`, will the SQL server returns a reasonable error?

Yeah, the server returns this error:
"""
failed to create instance sqlinstance-ssl-test: googleapi: Error 400: Invalid request: For a Postgres instance, sslMode value ENCRYPTED_ONLY and requireSsl value true are conflicting. When sslMode=ENCRYPTED_ONLY, requireSsl must be false. When requireSsl=true, sslMode must be TRUSTED_CLIENT_CERTIFICATE_REQUIRED. It's recommended that you only set sslMode.
"""

> * `ssLmode: ENCRYPTED_ONLY` and `requireSsl` unset (not configured in `spec`),  will KCC or the SQL server returns any message? I think we shall allow this case, that better aligns with the "cleared" concept and shall be covered by this test.

We do allow this case; it works as expected. The limitation is from the GCP API, not from direct controller / KCC validation. The reason I added this test case (with legacy field `requireSsl: false`) was to highlight this potential mistake a user could make. The API quirks are documented here: https://cloud.google.com/sql/docs/mysql/admin-api/rest/v1/instances#ipconfiguration

Also just updated the test case to remove `sslMode` from create.yaml, and remove `requireSsl` from update.yaml.

Reply:
Cool. 
For unset, I think what we want to verify is that create.yaml has `requireSsl: true` and `ssLmode: <valid value>` , the then update.yaml unset the `requireSsl`, this should pass because the update  "unset" the `requireSsl` rather than "unchange" it.

Reply:
Ok, updated!